app/services/foreman_ansible/roles_importer.rb

```diff
@@ -26,11 +27,15 @@ def import_roles(roles)
 
     def detect_changes(imported)
       changes = {}.with_indifferent_access
-      old, changes[:new] = imported.partition { |role| role.id.present? }
-      changes[:obsolete] = ::AnsibleRole.where.not(:id => old.map(&:id))
+      changes[:old], changes[:new] = imported.partition { |role| role.id.present? }
+      changes[:obsolete] = ::AnsibleRole.where.not(:id => changes[:old].map(&:id))
       changes
     end
 
+    def get_variables_names(role)
```

Comment:
Wouldn't this be better of in the `VariablesImporter`?
And it would probably be better to load variables for all the roles at once, as like this if we are importing higher number of roles, we make proxy API call per role. It is not very likely to be an issue, but there might be such cases.